### PR TITLE
Apply the Cedar schema terminology

### PR DESCRIPTION
**Pull request by shaobo-he-aws**
## Issue #, if available
#842 

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Requires updates, and I have made / will make these updates myself. (Please include in your description a timeline or link to the relevant PR in `cedar-spec`, and how you have tested that your updates are correct.)

### CODE REVIEW

**Comment by shaobo-he-aws** [comment]
All APIs and comments should be updated. The remaining work is to rename tests, which I will do once a preliminary review is done.

**Review by john-h-kastner-aws — COMMENTED** [review]
I like the proposed renaming.

Will this extend to renaming the policy format?

File: cedar-policy-cli/src/lib.rs [hunk 1]
```diff
@@ -140,23 +140,23 @@ pub struct TranslateSchemaArgs {
...
 #[derive(Debug, Clone, Copy, ValueEnum)]
 pub enum SchemaTranslationDirection {
-    /// JSON -> Human schema syntax
-    JsonToHuman,
-    /// Human schema syntax -> JSON
-    HumanToJson,
+    /// JSON -> Cedar schema syntax
+    JsonToCedar,
+    /// Cedar schema syntax -> JSON
+    CedarToJson,
```
Comment (cdisselkoen) [inline cedar-policy-cli/src/lib.rs:146]: This change should be called out in the CLI changelog

File: cedar-policy-cli/src/lib.rs [hunk 2]
```diff
@@ -140,23 +140,23 @@ pub struct TranslateSchemaArgs {
...
 }
 
 #[derive(Debug, Clone, Copy, ValueEnum)]
 pub enum SchemaFormat {
-    /// Human-readable format
-    Human,
+    /// the Cedar format
+    Cedar,
```
Comment (cdisselkoen) [inline cedar-policy-cli/src/lib.rs:152]: This change should be called out in the CLI changelog

**Review by cdisselkoen — APPROVED** [review]
Looks great

File: cedar-policy/src/api/err.rs [hunk 3]
```diff
@@ -161,17 +161,17 @@ impl From<cedar_policy_core::authorizer::ReauthorizationError> for Reauthorizati
     }
 }
 
-/// Errors serializing Schemas to the natural syntax
+/// Errors serializing Schemas to the Cedar syntax
 #[derive(Debug, Error, Diagnostic)]
-pub enum ToHumanSyntaxError {
+pub enum ToCedarSyntaxError {
```
Comment (khieta) [inline cedar-policy/src/api/err.rs:166]: I think the name `ToCedarSchemaError` would better convey that this error is schema specific

File: cedar-policy-validator/src/cedar_schema/parser.rs [hunk 4]
```diff
@@ -83,33 +83,33 @@ lazy_static::lazy_static! {
     static ref TYPE_PARSER: grammar::TypeParser = grammar::TypeParser::new();
 }
 
-/// Parse errors for parsing a human-syntax schema
+/// Parse errors for parsing a schema in the Cedar syntax
 #[derive(Debug, Diagnostic, Error)]
-pub enum HumanSyntaxParseErrors {
-    /// Parse error for the human syntax
+pub enum CedarSyntaxParseErrors {
```
Comment (khieta) [inline cedar-policy-validator/src/cedar_schema/parser.rs:88]: I think the name `CedarSchemaParseErrors` would better convey that this error is schema specific

File: cedar-policy-validator/src/err.rs [hunk 5]
```diff
@@ -38,21 +38,21 @@ pub enum HumanSchemaError {
...
-    Parsing(#[from] HumanSyntaxParseError),
+    Parsing(#[from] CedarSyntaxParseError),
 }
 
-/// Error parsing a human-syntax schema
+/// Error parsing a Cedar-syntax schema
 #[derive(Debug, Error)]
 #[error("error parsing schema: {errs}")]
-pub struct HumanSyntaxParseError {
+pub struct CedarSyntaxParseError {
```
Comment (khieta) [inline cedar-policy-validator/src/err.rs:47]: Same comment as above: I'd prefer `CedarSchemaParseError`

**Review by khieta — COMMENTED** [review]
Overall, the changes look good. The new names are more consistent.

But I'd prefer more symmetry between the public (aka defined in `api.rs`) `PolicySet` and `Schema` interfaces. For `PolicySet` we provide:
* a `FromStr` impl (which parses the Cedar syntax)
* `from_json_str`
* `from_json_value`
* `from_json_file`
* `to_json`

I think we should provide the same sort of interface for `Schema` and `SchemaFragment`. So:
* the `FromStr` impl should parse the Cedar syntax, and we should have a separate `from_json_str` to parse the JSON syntax
* `from_file` should be renamed `from_json_file`
* `from_file_cedar` should be renamed to `from_file`
* maybe  rename `as_cedar` to `to_cedar`

**Comment by shaobo-he-aws** [comment]
> I like the proposed renaming.
> 
> Will this extend to renaming the policy format?

Good idea. Will do.

**Comment by shaobo-he-aws** [comment]
> Overall, the changes look good. The new names are more consistent.
> 
> But I'd prefer more symmetry between the public (aka defined in `api.rs`) `PolicySet` and `Schema` interfaces. For `PolicySet` we provide:
> 
> * a `FromStr` impl (which parses the Cedar syntax)
> * `from_json_str`
> * `from_json_value`
> * `from_json_file`
> * `to_json`
> 
> I think we should provide the same sort of interface for `Schema` and `SchemaFragment`. So:
> 
> * the `FromStr` impl should parse the Cedar syntax, and we should have a separate `from_json_str` to parse the JSON syntax
> * `from_file` should be renamed `from_json_file`
> * `from_file_cedar` should be renamed to `from_file`
> * maybe  rename `as_cedar` to `to_cedar`

How about using the file suffix (`cedarschema` vs `json`) to name these APIs? For instance,

* `Schema::from_str_cedar` to `Schema::from_cedarschema_str`
* `Schema::from_file_cedar` to `Schema::from_cedarschema_file`
* `SchemaFragment::as_cedar` to `SchemaFragment::to_cedarschema`

File: cedar-policy/src/api.rs [hunk 6]
```diff
@@ -1338,7 +1340,7 @@ impl FromStr for Schema {
     /// found to not be a valid attribute name according to the Cedar
     /// grammar.
     fn from_str(schema_src: &str) -> Result<Self, Self::Err> {
-        Ok(Self(schema_src.parse()?))
+        Self::from_cedarschema_str(schema_src).map(|(schema, _)| schema)
     }
```
Comment (khieta) [inline cedar-policy/src/api.rs:1344]: This change makes sense to me, but I think the implementation for `SchemaFragment` still uses JSON. Make sure the two are consistent.

File: cedar-policy-validator/src/json_schema.rs [hunk 7]
```diff
@@ -152,20 +152,20 @@ impl Fragment<RawName> {
...
         file.read_to_string(&mut src)?;
-        Self::from_str_cedar(&src, extensions)
+        Self::from_cedarschema_str(&src, extensions)
     }
 }
 
 impl<N: Display> Fragment<N> {
     /// Pretty print this [`Fragment`]
-    pub fn as_cedar_schema(&self) -> std::result::Result<String, ToCedarSchemaSyntaxError> {
+    pub fn to_schemaschema(&self) -> std::result::Result<String, ToCedarSchemaSyntaxError> {
```
[the remark this answers is not in the document]
Comment (shaobo-he-aws) [reply cedar-policy-validator/src/json_schema.rs:168]: LOL. Good catch.

File: cedar-policy/src/api.rs (hunk 6 above)
Comment (shaobo-he-aws) [reply cedar-policy/src/api.rs:1346]: Good catch.

File: cedar-policy/src/api/err.rs (hunk 3 above)
Comment (shaobo-he-aws) [reply cedar-policy/src/api/err.rs:166]: Done.

File: cedar-policy-validator/src/cedar_schema/parser.rs (hunk 4 above)
Comment (shaobo-he-aws) [reply cedar-policy-validator/src/cedar_schema/parser.rs:88]: Done.

File: cedar-policy-validator/src/err.rs (hunk 5 above)
Comment (shaobo-he-aws) [reply cedar-policy-validator/src/err.rs:47]: Done.

File: cedar-policy/CHANGELOG.md [hunk 8]
```diff
@@ -52,6 +52,10 @@ Cedar Language Version: 4.0
 - (*) Implemented [RFC 52](https://github.com/cedar-policy/rfcs/blob/main/text/0052-reserved-namespaces.md).
   Names containing `__cedar` (e.g., `__cedar`, `A::__cedar`, `__cedar::A`, and
 `A::__cedar::B`) are now invalid. (#969)
+- Replace uses of "natural", "human", "human-readable", and "custom" with "Cedar" (#1114).
+  APIs with these names are changed accordingly. E.g., `Schema::from_str_natural` to `Schema::from_cedarschema_str`.
+  Moreover, the `FromStr` implementations of `Schema` and `SchemaFragment`
+  now parse strings in the Cedar schema format.
```
Comment (cdisselkoen) [inline cedar-policy/CHANGELOG.md:58]: nit: maybe add that you can still parse the JSON schema format using [method name]

File: cedar-policy/src/api.rs [hunk 9]
```diff
@@ -1220,8 +1220,18 @@ impl SchemaFragment {
         })
     }
 
+    /// Create a [`SchemaFragment`] from a string containing JSON in the
+    /// appropriate shape.
```
Comment (cdisselkoen) [inline cedar-policy/src/api.rs:1224]:
```suggestion
    /// Create a [`SchemaFragment`] from a string containing JSON in the
    /// JSON schema format.
```

File: cedar-policy/src/api.rs [hunk 10]
```diff
@@ -1264,28 +1276,30 @@ impl SchemaFragment {
         ))
     }
 
-    /// Create a [`SchemaFragment`] directly from a file.
-    pub fn from_file(file: impl std::io::Read) -> Result<Self, SchemaError> {
-        let lossless = cedar_policy_validator::json_schema::Fragment::from_file(file)?;
+    /// Create a [`SchemaFragment`] directly from a JSON file (which should
+    /// contain an object of the shape required for for the JSON schema format).
```
Comment (cdisselkoen) [inline cedar-policy/src/api.rs:1280]:
```suggestion
    /// Create a [`SchemaFragment`] directly from a JSON file (which should
    /// contain an object of the shape required for the JSON schema format).
```

File: cedar-policy/src/api.rs [hunk 11]
```diff
@@ -1338,12 +1348,12 @@ impl FromStr for Schema {
...
     fn from_str(schema_src: &str) -> Result<Self, Self::Err> {
-        Ok(Self(schema_src.parse()?))
+        Self::from_cedarschema_str(schema_src).map(|(schema, _)| schema)
     }
 }
 
 impl Schema {
-    /// Create a `Schema` from multiple `SchemaFragment`. The individual
+    /// Create a [`Schema`] from multiple [`SchemaFragment`]. The individual
     /// fragments may references entity types that are not declared in that
```
Comment (cdisselkoen) [inline cedar-policy/src/api.rs:1357]:
```suggestion
    /// fragments may reference entity or common types that are not declared in that
```

File: cedar-policy/src/api.rs [hunk 12]
```diff
@@ -1338,12 +1348,12 @@ impl FromStr for Schema {
...
     /// fragment, but all referenced entity types must be declared in some
```
Comment (cdisselkoen) [inline cedar-policy/src/api.rs:1358]:
```suggestion
    /// fragment, but all referenced entity and common types must be declared in some
```

**Review by cdisselkoen — APPROVED** [review]
Looks great

File: cedar-policy-cli/src/lib.rs (hunk 1 above)
Comment (shaobo-he-aws) [reply cedar-policy-cli/src/lib.rs:146]: Done.

File: cedar-policy-cli/src/lib.rs (hunk 2 above)
Comment (shaobo-he-aws) [reply cedar-policy-cli/src/lib.rs:152]: Done.

File: cedar-policy/CHANGELOG.md (hunk 8 above)
Comment (shaobo-he-aws) [reply cedar-policy/CHANGELOG.md]: Done.

File: cedar-policy/CHANGELOG.md [hunk 13]
```diff
@@ -52,6 +52,9 @@ Cedar Language Version: 4.0
 - (*) Implemented [RFC 52](https://github.com/cedar-policy/rfcs/blob/main/text/0052-reserved-namespaces.md).
   Names containing `__cedar` (e.g., `__cedar`, `A::__cedar`, `__cedar::A`, and
 `A::__cedar::B`) are now invalid. (#969)
+- Replace naming "natural", "human", "human-readable", and "custom" with "Cedar" (#1114).
+  APIs with these names (e.g., ) are changed accordingly. CLI arguments `--policy-format`
+  and `--schema-format` now take options `cedar` or `json`.
```
[the remark this answers is not in the document]
Comment (shaobo-he-aws) [reply cedar-policy/CHANGELOG.md]: Done.

File: cedar-policy-validator/src/schema.rs [hunk 14]
```diff
@@ -1126,7 +1126,7 @@ pub(crate) mod test {
     use super::*;
 
     /// Transform the output of functions like
-    /// `ValidatorSchema::from_str_natural()`, which has type `(ValidatorSchema, impl Iterator<...>)`,
+    /// `ValidatorSchema::from_str_cedar()`, which has type `(ValidatorSchema, impl Iterator<...>)`,
```
Comment (khieta) [inline cedar-policy-validator/src/schema.rs:1129]:
```suggestion
    /// `ValidatorSchema::from_cedarschema_str()`, which has type `(ValidatorSchema, impl Iterator<...>)`,
```

File: cedar-policy/src/api.rs [hunk 15]
```diff
@@ -1248,14 +1258,15 @@ impl SchemaFragment {
...
-    ) -> Result<(Self, impl Iterator<Item = SchemaWarning>), HumanSchemaError> {
-        let (lossless, warnings) = cedar_policy_validator::json_schema::Fragment::from_str_natural(
-            src,
-            Extensions::all_available(),
-        )?;
+    ) -> Result<(Self, impl Iterator<Item = SchemaWarning>), CedarSchemaError> {
+        let (lossless, warnings) =
+            cedar_policy_validator::json_schema::Fragment::from_cedarschema_str(
+                src,
+                Extensions::all_available(),
```
Comment (khieta) [inline cedar-policy/src/api.rs:1268]: Was about to comment here that the lossless schema should be the input `src`, but then realized that this is just an issue with the implementation, so I opened an issue instead #1118

**Review by khieta — APPROVED** [review]
Looks good, aside from a small typo.

**Review by khieta — APPROVED** [review]

**Pull request merged**